### PR TITLE
Improved Azure DevOps Date Handling

### PR DESCRIPTION
**Pull request by awanlin**
## Hey, I just made a Pull Request!

This change moves to using strings for Dates and letting the frontend handle the conversion to DateTime. We have been seeing odd issues with dates since the first PR for the backend plugin. This greatly improves the reliability of dates between the frontend and backend.

**Note:** This PR crosses 3 plugins/packages - azure-devops, azure-devops-backend, and azure-devops-common. This is the first time I've done this, hopefully this is possible

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by awanlin** [comment]
@marleypowell here's the PR we have been taking about. Please review when you have time. Also, should this be patch or minor?

**Review by marleypowell — APPROVED** [review]
LGTM 🙂

**Comment by awanlin** [comment]
@freben or any of the other maintainers, can you take a look at this PR, please? Would like your thoughts on this and also be able to unblock the other PRs for Azure DevOps we have in the works. Thanks in advance! 😃

File: .changeset/tame-buckets-move.md
Comment (benjdlambert) [inline .changeset/tame-buckets-move.md:4]: From what I can see I think that this is a `minor` change for the `common` package

Comment (awanlin) [reply .changeset/tame-buckets-move.md]: Sounds good, bit of a learning experience for me. I've made this change, thanks 👍

File: plugins/azure-devops/src/components/BuildTable/BuildTable.tsx
Comment (benjdlambert) [inline plugins/azure-devops/src/components/BuildTable/BuildTable.tsx:155]: Might be worth looking at this ADR for when displaying dates in the frontend to get them all standardized. https://backstage.io/docs/architecture-decisions/adrs-adr012

Comment (marleypowell) [reply plugins/azure-devops/src/components/BuildTable/BuildTable.tsx:155]: Don't think this covers relative times such as `10 seconds ago` but thanks for sharing the ADR 😄

Comment (awanlin) [reply plugins/azure-devops/src/components/BuildTable/BuildTable.tsx:155]: Thanks for pointing out the ADR, will keep that in mind, in this case as Marley mentioned we are using relative times so that way you can quickly see how old the builds are

Comment (benjdlambert) [reply plugins/azure-devops/src/components/BuildTable/BuildTable.tsx:155]: ahha - ok I see it now.

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**
